app/main/views/frameworks.py

```diff
@@ -68,21 +68,23 @@ def framework_dashboard(framework_slug):
             )
 
     drafts, complete_drafts = get_drafts(data_api_client, framework_slug)
-
     supplier_framework_info = get_supplier_framework_info(data_api_client, framework_slug)
     declaration_status = get_declaration_status_from_info(supplier_framework_info)
-    supplier_is_on_framework = get_supplier_on_framework_from_info(supplier_framework_info)
-
     # Do not show a framework dashboard for earlier G-Cloud iterations
     if declaration_status == 'unstarted' and framework['status'] == 'live':
         abort(404)
 
-    application_made = supplier_is_on_framework or (len(complete_drafts) > 0 and declaration_status == 'complete')
+    supplier_is_on_framework = get_supplier_on_framework_from_info(supplier_framework_info)
+    application_made = complete_drafts and declaration_status == 'complete'
     lots_with_completed_drafts = [lot for lot in framework['lots'] if count_drafts_by_lot(complete_drafts, lot['slug'])]
 
     framework_dates = content_loader.get_message(framework_slug, 'dates')
     framework_urls = content_loader.get_message(framework_slug, 'urls')
 
+    # Confidence banner.
+    if application_made and framework['status'] == 'open':
+        flash('submission_confidence', 'success')
```

Comment:
Yeah, in general we flash a 'key' that corresponds to a message value in the template for banners.
Thought it was a bit weird at first but it does allow quite nicely for the separation of logic and content, although going forward it would be nice if we had some sort of location for all key: message/ value pairs.